src/hooks/useMediaQuery.ts

```diff
@@ -0,0 +1,37 @@
+import { useEffect, useLayoutEffect, useState } from "react"
+
+
+const getMatches = (query: string) => {
+  // Prevents SSR issues
+  if (typeof window !== "undefined") {
+    return window.matchMedia(query).matches
+  }
+
+  return false
+}
+
+export const useMediaQuery = (query: string, layout = false) => {
+  const [matches, setMatches] = useState(getMatches(query))
+  const effectFunc = layout ? useLayoutEffect : useEffect
```

Comment:
Blæh, litt usikker på dette. Er det ikke en `rule of hooks` som sier at man aldri skal kalle på hooks conditionally? En liten frykt for at dette kan føre til uheldige side-effects, men kan hende at jeg er litt vel forsiktig

Reply:
Hehe jeg tenkte du kom til å reagere på det. Ja, man må kjøre alle hooks i samme rekkefølge hver gang - og det vil denne gjøre så lenge du ikke gjør dynamiske endringer på `layout`. Jeg mener at "layout" er tett nok knyttet til et velkjent React-konsept til at det ikke bør være store problemer her.

Forslag: Options-objekt heller? Mer vekt på at det er config. Naturlig å la de andre hooksene ta samme options og passe videre. Merkes det at jeg liker å lage library kode? 🙃

Reply:
Kanskje tilogmed kalle den `layoutEffect:` eller bare straight up gi funksjonen som option.

Reply:
Hva med å gjøre hele `layout` til en intern greie? Så kan man heller eksportere `useLayoutMediaQuery` og `useMediaQuery`. Da blir det vanskelig å sende inn noe reaktivt

Reply:
Ja, men da må du lage dobbelt opp alle steder du vil bygge småhooks oppå.

Reply:
Kunne puttet `layout` i en `useRef(layout)`, så forsikrer man seg for at den ikke bytter verdi?

Reply:
Da kan du ikke bruke den på render

Reply:
Tenkte noe ala dette:
```
export default function useConstant<T>(v:  T): T {
  const ref = useRef<ResultBox<T>>()

  if (!ref.current) {
    ref.current = { v: v }
  }

  return ref.current.v
}
```

Men blir kanskje litt  overengeneering 😅